### PR TITLE
feat(shell-api): support `type` field for search index creation MONGOSH-1631

### PR DESCRIPTION
**Pull request by addaleax**
Now that SERVER-83107 is complete, we can add this to mongosh.

### CODE REVIEW

File: packages/shell-api/src/collection.spec.ts [hunk 1]
```diff
@@ -2584,6 +2584,61 @@ describe('Collection', function () {
...
+                type: 'vectorSearch',
+                definition: { mappings: { dynamic: true } },
+              },
+            ]
+          );
+        });
+      });
+
+      context('with name, options and type === search', function () {
+        it('calls serviceProvider.createIndexes', async function () {
```
Comment (lerouxb) [inline packages/shell-api/src/collection.spec.ts:2609]: serviceProvider.createSearchIndexes?

File: packages/shell-api/src/collection.spec.ts [hunk 2]
```diff
@@ -2584,6 +2584,61 @@ describe('Collection', function () {
           );
         });
       });
+
+      context('with name, options and type !== search', function () {
+        it('calls serviceProvider.createIndexes', async function () {
```
Comment (lerouxb) [inline packages/shell-api/src/collection.spec.ts:2589]: serviceProvider.createSearchIndexes?

File: packages/shell-api/src/collection.spec.ts [hunk 3]
```diff
@@ -2584,6 +2584,61 @@ describe('Collection', function () {
...
+          expect(serviceProvider.createSearchIndexes).to.have.been.calledWith(
+            'db1',
+            'coll1',
+            [{ name: 'my-index', definition: { mappings: { dynamic: true } } }]
+          );
+        });
+      });
+
+      context('with options and type but no name', function () {
+        it('calls serviceProvider.createIndexes', async function () {
```
Comment (lerouxb) [inline packages/shell-api/src/collection.spec.ts:2623]: and here

**Review by lerouxb — APPROVED** [review]

File: packages/shell-api/src/collection.spec.ts (hunk 2 above)
Comment (addaleax) [reply packages/shell-api/src/collection.spec.ts:2589]:
```suggestion
        it('calls serviceProvider.createSearchIndexes', async function () {
```

**Pull request merged**